sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/frequency/KMostFrequent.java

```diff
@@ -0,0 +1,200 @@
+ * https://pdfs.semanticscholar.org/72f1/5aba2e67b1cc9cd1fb12c99e101c4c1aae4b.pdf
+ * <br>The implementation comes from Addthis' library Stream-lib : https://github.com/addthis/stream-lib
+ */
+public class KMostFrequent {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KMostFrequent.class);
+
+    // do not instantiate
+  private KMostFrequent() {
+  }
+
+  /**
+   * A {@code PTransform} that takes an inputT {@code PCollection} of objects and returns a
+   * {@code PCollection<StreamSummary>} whose contents is a sketch which contains
+   * the most frequent elements in the input {@code PCollection}.
+   *
+   * <p>The {@code capacity} parameter controls the maximum number of elements the sketch
+   * can contain. Once this capacity is reached, the least frequent element is dropped each
+   * time an incoming element is not already present in the sketch.
+   * Each element in the sketch is associated to a counter, that keeps track of the estimate
+   * frequency as well as the maximal potential error.
+   * <br>See {@link KMostFrequentFn} for more details.
+   *
+   * <p>Example of use
+   * <pre> {@code PCollection<String> input = ...;
+   * PCollection<StreamSummary<String>> ssSketch = input
+   *        .apply(SketchFrequencies.<String>perKey(10000));
+   * } </pre>
+   *
+   * @param capacity    the maximum number off distinct elements that the Stream Summary can keep
+   *                    track of at the same time
+   * @param <InputT>    the type of the elements in the input {@code PCollection}
+   */
+  public static <InputT> Combine.Globally<InputT, StreamSummary<InputT>> globally(int capacity) {
+    return Combine.<InputT, StreamSummary<InputT>>globally(new KMostFrequentFn<InputT>(capacity));
+  }
+
+  /**
+   * A {@code PTransform} that takes an input {@code PCollection<KV<K, InputT>>} and returns a
+   * {@code PCollection<KV<K, StreamSummary>>} that contains an output element mapping each
+   * distinct key in the input {@code PCollection} to a sketch which contains the most frequent
+   * values associated with that key in the input {@code PCollection}.
+   *
+   * <p>The {@code capacity} parameter controls the maximum number of elements the sketch
+   * can contain. Once this capacity is reached, the least frequent element is dropped each
+   * time an incoming element is not already present in the sketch.
+   * Each element in the sketch is associated to a counter, that keeps track of the estimate
+   * frequency as well as the maximal potential error.
+   * <br>See {@link KMostFrequentFn} for more details.
+   *
+   * <p>Example of use
+   * <pre> {@code PCollection<KV<Integer, String>> input = ...;
+   * PCollection<KV<Integer, StreamSummary<String>>> ssSketch = input
+   *        .apply(SketchFrequencies.<Integer, String>globally(10000));
+   * } </pre>
+   *
+   * @param capacity    the maximum number off distinct elements that the Stream Summary can keep
```

Comment:
<!--new_thread; commit:6bf51fdf6afe6f389415792f335dd08adf9fa85a; resolved:0-->
`off -> of`